src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -288,6 +289,7 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             }
             _reloadEnvironmentVariables = winrt::unbox_value_or<bool>(settings.TryLookup(L"reloadEnvironmentVariables").try_as<Windows::Foundation::IPropertyValue>(),
                                                                       _reloadEnvironmentVariables);
+            _profileGuid = winrt::unbox_value_or<winrt::guid>(settings.TryLookup(L"profileGuid").try_as<Windows::Foundation::IPropertyValue>(), _profileGuid);
```

Comment:
you don't need to do anything about this, i just felt like I had a duty to say it

Reply:
Hmmmm... now that we have the `WSLENV` dance happening over all provided environment variables in `ConptyConnection`, do you think that we _could_ get away with moving profile GUID up to TerminalPage?

It could become annoying because now we need to copy the user's provided environment map (if there is one! but we do have to copy it because we don't want to edit it inplace since the one living on the profile is durable)... but after all, if TPage provides `WT_PROFILE_ID` via the environment map it will automatically show up in `WSLENV`

Reply:
I think that depends on if we think people will try and override that environment variable.

The keys in the `IMap` are case sensitive - so if someone has added say `wt_profile_id` as an environment variable, that'll be stored separately - so back in `ConptyConnection` one of the two values will be put into the connection's environment.

Reply:
Fair!
We can fix the interface later.

Reply:
One potential solution would be to pass _two_ environment variable maps to `ConptyConnection` - one containing user variables and one containing "terminal" variables. We'd insert the "terminal" ones into the environment first, and then the user ones. I can implement that as a follow-up if you're happy with that approach (and please say if you have a better name for "terminal" variables too!).